src/NanoTranscriber.cpp

```diff
@@ -0,0 +1,171 @@
+#include <sensor_msgs/Range.h>
+#include <iarc7_msgs/Nano.h>
+#include <iarc7_msgs/FlowVector.h>
+#include <iarc7_msgs/Float64Stamped.h>
+#include <iarc7_msgs/PlanarThrottle.h>
+#include <iarc7_msgs/ESCCommand.h>
+#include <iarc7_msgs/OrientationThrottleStamped.h>
+#include <iarc7_msgs/FlightControllerStatus.h>
+#include <sensor_msgs/Joy.h>
+#include <ros/ros.h>
+
+void distributeMessages(iarc7_msgs::Nano nano_info, 
+                        ros::Publisher& opticalflow_pub,
+                        ros::Publisher& rangefinder_pub,
+                        ros::Publisher& long_range_pub,
+                        ros::Publisher& battery_publisher){
+
+    // From here on out, each of the "offset" variables is multiplied by 10000 to convert 
+    // from the tens of microseconds offsets given to us by the AVR chip to nanoseconds.
+
+    sensor_msgs::Range long_range_msg;
+    sensor_msgs::Range short_range_msg;
+    iarc7_msgs::FlowVector flow_msg;
+    iarc7_msgs::Float64Stamped battery_voltage;
+
+    long_range_msg.range = nano_info.long_range;
+    long_range_msg.min_range = 0.5;
+    long_range_msg.max_range = 10;
+    ros::Duration long_range_offset = ros::Duration().fromNSec((int64_t)(nano_info.long_range_offset * 10000));
+    long_range_msg.header.stamp = nano_info.msg_received - long_range_offset;
+    long_range_msg.radiation_type = sensor_msgs::Range::INFRARED;
+    long_range_msg.field_of_view = 0.3;
+    long_range_msg.header.frame_id = "/tfmini";
+    if(nano_info.long_range_offset > 0) {
```

Comment:
Does an offset of 0 mean that measurement is invalid?  If so can you put a comment saying that here?

Reply:
Yes, the offsets are set to 0 by default if the sensor is not ready with a measurement yet. I will add a comment there